frontend/src/pages/AuthPage.tsx

```diff
@@ -34,32 +34,34 @@ function AuthPage() {
   const { theme } = useContext(ThemeContext);
 
   const [session, setSession] = useState<Session | null>();
-  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const navigate = useNavigate();
 
   useEffect(() => {
     supabase.auth.getSession().then(({ data: { session } }) => {
       setSession(session);
       if (session) {
-        setIsLoggedIn(true);
+        console.log('User is logged in');
       }
     });
 
     const {
       data: { subscription }
-    } = supabase.auth.onAuthStateChange((_event, session) => {
+    } = supabase.auth.onAuthStateChange((event, session) => {
       setSession(session);
-      if (session) {
-        setIsLoggedIn(true);
+      if (event == 'SIGNED_IN') {
+        // console.log("signed in");
+        // console.log(user);
+        navigate('/');
       }
     });
 
     return () => subscription.unsubscribe();
-  }, []);
+  }, [navigate]);
```

Comment:
hmm i don't really get this HAHA

Reply:
oh, the box is just the dependencies i think

Reply:
i don't think so because if you change this to session or empty or some random variable it still works